Fix #3651: [RTL] High-fi rotation animation for icon in Topic lesson and Hints and Solution

veena14cs opened the pull request:
## Explanation

This PR fixes #3651. This PR fixes expand/collapse icon for RTL.

## Test with new RTL code passes as expected.
<img width="384" alt="Screenshot (57)" src="https://user-images.githubusercontent.com/16301028/128565069-4ecb117d-00a3-4f2e-ab94-104c1e0b683a.png">


## Test without new RTL codes fails as expected.

<img width="384" alt="Screenshot (60)" src="https://user-images.githubusercontent.com/16301028/128606354-6caea2dc-22f5-45f9-ac74-32bd757a9d1d.png">




## Checklist
<!-- Please tick the relevant boxes by putting an "x" in them. -->
- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

veena14cs committed:
Update selection_interaction_item.xml

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Update selection_interaction_item.xml

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Update return_to_topic_button_item.xml

veena14cs committed:
Update return_to_topic_button_item.xml

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Fixed rotation animation for icon in Topic lesson and Hints and Solution

veena14cs committed:
Optimized code

veena14cs committed:
Update ViewBindingAdaptersTest.kt

veena14cs committed:
Merge branch 'develop' into fix-icon-animation

veena14cs committed:
Fixed tests

rt4914 reviewed (changes requested):
@veena14cs Can you please self review this PR? Looks like there as some unwanted changes also the branch is not up-to-date with develop.

anandwana001 commented on app/src/main/AndroidManifest.xml:
```diff
@@ -5,7 +5,7 @@
 
   <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE" />
   <uses-permission android:name="android.permission.INTERNET" />
-  <!-- TODO(#56): Reenable landscape support. -->
```
why we are removing this?

veena14cs replied:
Reverted.

anandwana001 commented on app/src/main/AndroidManifest.xml:
```diff
@@ -238,4 +240,4 @@
       tools:node="remove" />
   </application>
-</manifest>
+</manifest>
```
EOF

veena14cs replied:
Done.

anandwana001 commented on app/src/main/res/layout/activity_view_binding_adapters_test.xml:
```diff
@@ -0,0 +1,20 @@
+      android:padding="8dp"
+      android:src="@drawable/ic_arrow_drop_down_black_24dp" />
+  </FrameLayout>
+</layout>
```
EOF

veena14cs replied:
Done.

anandwana001 commented on app/src/sharedTest/java/org/oppia/android/app/databinding/ViewBindingAdaptersTest.kt:
```diff
@@ -0,0 +1,245 @@
+import javax.inject.Inject
+import javax.inject.Singleton
+
+private const val TOLERANCE = 1e-5f
```
add kdoc

veena14cs replied:
Done.

anandwana001 reviewed (changes requested):
Thanks, @veena14cs 

I haven't reviewed it in detail, looks like few changes. I will check the tests tomorrow.

veena14cs committed:
Fixed unwanted code changes

veena14cs committed:
Update accessibility_label_exemptions.textproto

veena14cs commented:
> Thanks, @veena14cs
> 
> I haven't reviewed it in detail, looks like few changes. I will check the tests tomorrow.

Thanks @anandwana001 I have addressed your comments PTAL.

veena14cs commented:
> @veena14cs Can you please self review this PR? Looks like there as some unwanted changes also the branch is not up-to-date with develop.

Fixed unwanted changes PTAL.

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge branch 'develop' into fix-icon-animation

rt4914 commented on app/src/main/java/org/oppia/android/app/databinding/ViewBindingAdapters.java:
```diff
@@ -45,23 +44,25 @@ public static void setLayoutWidth(@NonNull View view, float width) {
+    return ViewCompat.getLayoutDirection(view) == ViewCompat.LAYOUT_DIRECTION_RTL;
+  }
 }
+
```
Remove this extra line.

veena14cs replied:
Done.

rt4914 commented on app/src/main/res/layout/activity_view_binding_adapters_test.xml:
```diff
@@ -0,0 +1,20 @@
+    android:layout_marginStart="8dp"
+    android:layout_marginEnd="8dp"
+    android:layout_marginBottom="8dp"
+    >
```
Press `Control + Alt + L` - reformatting shortcut to fix the formatting.

veena14cs replied:
Done.

rt4914 reviewed (approved):
Suggested nit changes. Otherwise LGTM

veena14cs committed:
fixed nit

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge branch 'develop' into fix-icon-animation

anandwana001 reviewed (approved):
LGTM

BenHenning commented:
Hi. Will need to look at this tomorrow.

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge branch 'develop' into fix-icon-animation

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge branch 'develop' into fix-icon-animation

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge branch 'develop' into fix-icon-animation

BenHenning reviewed (approved):
Thanks @veena14cs. I'm glad we were able to effectively test this. PR LGTM!
